### PR TITLE
New perf metric - e2e throughput

### PR DESCRIPTION
**Pull request by ytaous**
**Description**: Adding new throughput metric for reporting.

**Motivation and Context**
- We'll switch to the new throughput for dashboard monitoring.

### CODE REVIEW

File: orttraining/orttraining/models/runner/training_runner.cc [hunk 1]
```diff
@@ -923,6 +923,15 @@ Status TrainingRunner::TrainingLoop(IDataLoader& training_data_loader, IDataLoad
     ++epoch;
   }
 
+  double e2e_throughput{0};
```
Comment (edgchen1) [inline orttraining/orttraining/models/runner/training_runner.cc:926]: e2e_throughput can be const:
```cpp
const double e2e_throughput = [&]() {
  if (end_to_end_perf_start_step >= params_.num_train_steps) return 0;
  ...
  return params_.batch_size * end_to_end_step_count / total_e2e_time;
}();
```

File: tools/perf_util/src/main/java/com/msft/send_perf_metrics/App.java [hunk 2]
```diff
@@ -98,7 +98,7 @@ static private void loadMetricsIntoMySQL(java.sql.Connection conn, String commit
 		// field name -> json value
 		Map<String, Object> field_mapping = new LinkedHashMap();
 		Set<String> update_on_duplicate_fields =
-			new LinkedHashSet<> (Arrays.asList("AvgTimePerBatch", "Throughput", "StabilizedThroughput", "TotalTime", "AvgCPU", "Memory"));
+			new LinkedHashSet<> (Arrays.asList("AvgTimePerBatch", "Throughput", "StabilizedThroughput", "EndToEndThroughput","TotalTime", "AvgCPU", "Memory"));
```
Comment (edgchen1) [inline tools/perf_util/src/main/java/com/msft/send_perf_metrics/App.java:101]: space after "EndToEndThroughput"
nice to see that adding a new field here is simple :)

**Comment by yuslepukhin** [comment]
> 		ProcessBuilder sb = new ProcessBuilder(commands).directory(source_dir.toFile()).redirectErrorStream(true);

Why all the tabs? I think we practice spaces

---
Refers to: tools/perf_util/src/main/java/com/msft/send_perf_metrics/App.java:22 in 72cf569. [](commit_id = 72cf5697c810a9fe34f5948aaed1f3bcf50c4f77, deletion_comment = False)

**Comment by ytaous** [comment]
> 		ProcessBuilder sb = new ProcessBuilder(commands).directory(source_dir.toFile()).redirectErrorStream(true);

this is very strange, my intellij's setting, the tab is unchecked, it should put 4 spaces for each tab. Luckily there's shortcut to replace all tabs with spaces.
Thanks for pointing this out.

---
In reply to: [636217921](https://github.com/microsoft/onnxruntime/pull/4085#issuecomment-636217921) [](ancestors = 636217921)

---
Refers to: tools/perf_util/src/main/java/com/msft/send_perf_metrics/App.java:22 in 72cf569. [](commit_id = 72cf5697c810a9fe34f5948aaed1f3bcf50c4f77, deletion_comment = False)

**Pull request merged**